x/proof/keeper/msg_server_create_claim_test.go

```diff
@@ -508,6 +536,114 @@ func TestMsgServer_CreateClaim_Error(t *testing.T) {
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	srv := keeper.NewMsgServerImpl(*keepers.Keeper)
+
+	// The base session start height used for testing
+	sessionStartHeight := int64(1)
+	// service is the only service for which a session should exist.
+	// this service has a value of greater than 1 for the compute units per relay.
+	service := &sharedtypes.Service{
+		Id:                   testServiceId,
+		ComputeUnitsPerRelay: nonDefaultComputeUnitsPerRelay,
+		OwnerAddress:         sample.AccAddress(),
+	}
+	// supplierAddr is staked for "svc1" such that it is expected to be in the session.
+	supplierAddr := sample.AccAddress()
+
+	// appAddr is staked for "svc1" such that it is expected to be in the session.
+	appAddr := sample.AccAddress()
+
+	supplierKeeper := keepers.SupplierKeeper
+	appKeeper := keepers.ApplicationKeeper
+
+	// Add a supplier that is expected to be in the session.
+	supplierKeeper.SetSupplier(ctx, sharedtypes.Supplier{
+		OperatorAddress: supplierAddr,
+		Services: []*sharedtypes.SupplierServiceConfig{
+			{Service: service},
+		},
+	})
+
+	// Add an application that is expected to be in the session.
+	appKeeper.SetApplication(ctx, apptypes.Application{
+		Address: appAddr,
+		ServiceConfigs: []*sharedtypes.ApplicationServiceConfig{
+			{Service: service},
+		},
+	})
+
+	// Add the service that is expected to be on-chain.
+	keepers.SetService(ctx, *service)
+
+	// Query for the session which contains the expected app and supplier pair.
+	sessionRes, err := keepers.SessionKeeper.GetSession(
+		ctx,
+		&sessiontypes.QueryGetSessionRequest{
+			ApplicationAddress: appAddr,
+			Service:            service,
+			BlockHeight:        1,
+		},
+	)
+	require.NoError(t, err)
+	require.NotNil(t, sessionRes)
+	require.Equal(t, appAddr, sessionRes.GetSession().GetApplication().GetAddress())
+
+	sessionResSuppliers := sessionRes.GetSession().GetSuppliers()
+	require.NotEmpty(t, sessionResSuppliers)
+	require.Equal(t, supplierAddr, sessionResSuppliers[0].GetOperatorAddress())
+
+	// Increment the block height to the test claim height.
+	sessionHeader := sessionRes.GetSession().GetHeader()
+	sharedParams := keepers.SharedKeeper.GetParams(ctx)
+	testClaimHeight := shared.GetClaimWindowCloseHeight(&sharedParams, sessionHeader.GetSessionEndBlockHeight())
+	sdkCtx = sdkCtx.WithBlockHeight(testClaimHeight)
+	ctx = sdkCtx
+
+	testClaimMsg := newTestClaimMsg(t,
+		sessionStartHeight,
+		sessionRes.GetSession().GetSessionId(),
+		supplierAddr,
+		appAddr,
+		service,
+		defaultMerkleRoot,
+	)
+
+	// use the test claim message to create a claim object to get the number of relays and compute units in the claim.
+	testClaim := types.Claim{RootHash: testClaimMsg.GetRootHash()}
+	testClaimNumComputeUnits, err := testClaim.GetNumComputeUnits()
+	require.NoError(t, err)
+	testClaimNumRelays, err := testClaim.GetNumRelays()
+	require.NoError(t, err)
+
+	createClaimRes, err := srv.CreateClaim(ctx, testClaimMsg)
+
```

Comment:
```suggestion
	// Ensure that submitting the claim fails because the number of compute units
	// claimed does not match the expected amount as a function of (relay, service_CUPR)
	createClaimRes, err := srv.CreateClaim(ctx, testClaimMsg)
```